src/common/components/Carousel/index.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  Children,
+  cloneElement,
+  ComponentProps,
+  isValidElement,
+  ReactNode,
+  useRef,
+} from 'react';
+
+import Group from '~/common/components/Group';
+
+import CarouselButton from './CarouselButton';
+import useDragScroll from './useDragScroll';
+
+interface CarouselProps extends ComponentProps<'div'> {
+  children: ReactNode;
+  itemsToShow?: number;
+  childSize?: number;
+  useButton?: boolean;
+  groupGap?: number;
+}
+
+const Carousel = ({
+  children,
+  itemsToShow = 4,
+  childSize = 100,
+  useButton = false,
+  groupGap = 5,
+  className,
+  ...props
+}: CarouselProps) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const itemToShowWidth = itemsToShow * (childSize + groupGap) + childSize / 2;
+  const totalCarousels = Children.count(children);
+
+  const avatars = Children.toArray(children).map((child: ReactNode) => {
+    if (isValidElement(child)) {
+      return cloneElement(child as JSX.Element, {
+        style: { width: `${childSize}px` },
+      });
+    }
+  });
+
+  const {
+    isLeftButtonActive,
+    isRightButtonActive,
+    buttonScrollLeft,
+    buttonScrollRight,
+    handleMouseDown,
+    handleMouseMove,
+    handleMouseUp,
+  } = useDragScroll({
+    containerRef,
+    childSize: childSize + groupGap,
+  });
+
+  return (
+    <div className="flex items-center" {...props}>
+      <div
+        ref={containerRef}
+        className={`order-2 flex snap-x overflow-hidden scroll-smooth py-2 ${className}`}
```

Comment:
className에 중복된 속성이 들어올 수 있으니 cn을 사용하는게 좋을 것 같아요!
```suggestion
        className={cn('order-2 flex snap-x overflow-hidden scroll-smooth py-2', className)}
```